modules/gdscript/gdscript_editor.cpp

```diff
@@ -2722,19 +2995,27 @@ static bool _guess_method_return_type_from_base(GDScriptParser::CompletionContex
 }
 
 static void _find_enumeration_candidates(GDScriptParser::CompletionContext &p_context, const String &p_enum_hint, HashMap<String, ScriptLanguage::CodeCompletionOption> &r_result) {
+	const HashMap<String, DocData::ClassDoc> class_list = EditorHelp::get_doc_data()->class_list;
 	if (!p_enum_hint.contains_char('.')) {
 		// Global constant or in the current class.
 		StringName current_enum = p_enum_hint;
 		if (p_context.current_class && p_context.current_class->has_member(current_enum) && p_context.current_class->get_member(current_enum).type == GDScriptParser::ClassNode::Member::ENUM) {
 			const GDScriptParser::EnumNode *_enum = p_context.current_class->get_member(current_enum).m_enum;
 			for (int i = 0; i < _enum->values.size(); i++) {
+				// TODO: Determine what case this covers.
```

Comment:
Enum members from the current GDScript, specific to a certain context. So when a type parameter of a function requires an enum this method gets called an will search possible enum values for this specifc enum. (Also happens with assignments IIRC)

Reply:
Ok so I looked again. For function calls it only gets called for functions of native types (e.g. `Control.set_anchor`). Might be something to extend to CLASS types though, since the function would be able to handle it from what I see. For assignments it could get called for class types, but in practice never is, since the type attached to gdscript variables doesn't have the enumeration name set (might be a bug don't know). So yeah for now it is only ever called for native functions and properties.